src/sentry/static/sentry/app/components/hovercard.jsx

```diff
@@ -1,162 +1,177 @@
-      timeout
-    );
+    this.hoverWait = setTimeout(() => this.setState({visible}), displayTimeout);
   };
 
-  positionCard() {
-    if (!this.cardElement.current || this.state.visible) {
-      return;
-    }
-    const rect = this.cardElement.current.getBoundingClientRect();
-
-    // Computes the offset that the hovercard should be from the anchor point
-    // of the container (top or bottom absolute position).
-    const offset = this.containerElement.current.offsetHeight;
-
-    const y = rect.top - offset < 0 ? 'bottom' : 'top';
-    const x =
-      rect.right > window.innerWidth && !(rect.left < 0)
-        ? 'right'
-        : rect.left < 0
-        ? 'left'
-        : 'middle';
-
-    this.setState({position: {x, y, offset}});
-  }
-
   render() {
-    const {bodyClassName, containerClassName, className, header, body} = this.props;
-    const {rendered, visible, position} = this.state;
+    const {
+      bodyClassName,
+      containerClassName,
+      className,
+      header,
+      body,
+      position,
+    } = this.props;
+    const {visible} = this.state;
 
     // Maintain the hovercard class name for BC with less styles
     const cx = classNames('hovercard', className);
+    const modifiers = {
+      hide: {
+        enabled: false,
+      },
+      preventOverflow: {
+        padding: 10,
+        enabled: true,
+        boundariesElement: 'viewport',
+      },
+    };
 
     return (
-      <Container
-        className={containerClassName}
-        innerRef={this.containerElement}
-        onMouseEnter={this.handleToggleOn}
-        onMouseLeave={this.handleToggleOff}
-      >
-        {this.props.children}
-        {rendered && (
-          <StyledHovercard
-            visible={visible}
-            withHeader={!!header}
-            className={cx}
-            position={position || DEFAULT_POSITION}
-            innerRef={this.cardElement}
-          >
-            {header && <Header>{header}</Header>}
-            {body && <Body className={bodyClassName}>{body}</Body>}
-          </StyledHovercard>
-        )}
-      </Container>
+      <Manager>
+        <Reference>
+          {({ref}) => (
+            <Container
+              innerRef={ref}
+              aria-describedby={this.tooltipId}
+              onMouseEnter={this.handleToggleOn}
+              onMouseLeave={this.handleToggleOff}
+              className={containerClassName}
+            >
+              {this.props.children}
+            </Container>
+          )}
+        </Reference>
+        {visible &&
+          ReactDOM.createPortal(
+            <Popper placement={position} modifiers={modifiers}>
+              {({ref, style, placement, arrowProps}) => (
+                <StyledHovercard
+                  innerRef={ref}
+                  visible={visible}
+                  withHeader={!!header}
+                  style={style}
+                  data-placement={placement}
```

Comment:
why `data-` attr instead of a prop?

Reply:
I thought the attributes would make CSS simpler as I wouldn't have to change contexts as often.

Reply:
@Chrissy any opinions here?

Reply:
i guess i'm not really sure how a data-attribute would help? I would prefer to avoid something like `&[data-placement='top'] { etc }` because it is hard to use composition with, has unreferenced dependencies etc. But Idunno if that is what you were thinking...

Reply:
I'll try re-jigging this to pass a prop to the container / arrow and do the directional offsets in the style blocks.